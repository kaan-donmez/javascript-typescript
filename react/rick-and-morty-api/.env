VITE_RICK_AND_MORTY_URL = "https://rickandmortyapi.com/graphql"