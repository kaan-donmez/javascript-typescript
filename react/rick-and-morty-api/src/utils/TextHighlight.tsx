function TextHighlight({ input, label }: { input: string; label: string }) {
  if (input.length === 0) return label;

  const pattern = new RegExp(`(${input})`, "gi");
  const parts = label.split(pattern);

  return parts.map((part, index) =>
    pattern.test(part) ? <strong key={index}>{part}</strong> : part
  );
}

export default TextHighlight;
