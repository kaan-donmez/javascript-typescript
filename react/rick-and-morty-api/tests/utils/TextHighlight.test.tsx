import { render } from "@testing-library/react";
import TextHighlight from "../../src/utils/TextHighlight";

describe("TextHighlight", () => {
  it("should render the TextHighlight component with proper input value", () => {
    const { asFragment } = render(
      <TextHighlight input="summ" label="Summer summer" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the TextHighlight component with empty input value", () => {
    const { asFragment } = render(<TextHighlight input="" label="Summer" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
