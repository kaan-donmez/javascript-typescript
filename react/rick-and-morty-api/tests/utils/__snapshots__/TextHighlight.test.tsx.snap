// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextHighlight > should render the TextHighlight component with empty input value 1`] = `
<DocumentFragment>
  Summer
</DocumentFragment>
`;

exports[`TextHighlight > should render the TextHighlight component with proper input value 1`] = `
<DocumentFragment>
  <strong>
    Summ
  </strong>
  er 
  <strong>
    summ
  </strong>
  er
</DocumentFragment>
`;
